### PR TITLE
[docs-only] Adjust idm docs after it's now running by default

### PR DESCRIPTION
**Pull request by rhafer**
Add some general words about idm and remove the setup steps that are no longer needed. Also update the service communication chart

### CODE REVIEW

File: docs/extensions/idm/_index.md [hunk 1]
```diff
@@ -10,6 +10,19 @@ geekdocCollapseSection: true
...
+It is mainly targeted at small oCIS installations. For larger setups it is recommended to replace IDM with a "real" LDAP server
+or to switch to an external Identity Management Solution.
+
+IDM listens on port 9325 by default. In the default configuration it only accepts TLS protected connections (LDAPS). The BaseDN
+of the LDAP tree is `o=libregraph-idm`. IDM gives LDAP write permissions to a single user 
+(DN: `uid=libregraph,ou=sysusers,o=libregraph-idm`) any other authenticated user has read-only access.
+
+Note: IDM is limited in its functionality. It only supports a subset of the LDAP operations (namely BIND, SEARCH, ADD, MODIFY, DELETE).
+Also IDM currently does not do any Schema Verification (e.g. structural vs. auxillary Objectclasses, require and option Attributes,
+Syntax Checks, ...). So it's not meant as a general purpose LDAP server.
```
Comment (micbar) [inline docs/extensions/idm/_index.md:25]: Can you add one sentence about the storage location of the boltdb please?

File: docs/extensions/idm/_index.md [hunk 2]
```diff
@@ -10,6 +10,20 @@ geekdocCollapseSection: true
 
 ## Abstract
 
+The IDM service provides a minimal LDAP Service (based on https://github.com/libregraph/idm) for oCIS. It is started as part of
+the default configuration and serves as a central place for storing user and group informationn.
```
Comment (micbar) [inline docs/extensions/idm/_index.md:14]:
```suggestion
the default configuration and serves as a central place for storing user and group information.
```

File: docs/extensions/idm/_index.md [hunk 3]
```diff
@@ -10,6 +10,20 @@ geekdocCollapseSection: true
...
+
+It is mainly targeted at small oCIS installations. For larger setups it is recommended to replace IDM with a "real" LDAP server
+or to switch to an external Identity Management Solution.
+
+IDM listens on port 9325 by default. In the default configuration it only accepts TLS protected connections (LDAPS). The BaseDN
+of the LDAP tree is `o=libregraph-idm`. IDM gives LDAP write permissions to a single user 
+(DN: `uid=libregraph,ou=sysusers,o=libregraph-idm`) any other authenticated user has read-only access. IDM stores its data in a
```
Comment (micbar) [inline docs/extensions/idm/_index.md:21]:
```suggestion
(DN: `uid=libregraph,ou=sysusers,o=libregraph-idm`). Any other authenticated user has read-only access. IDM stores its data in a
```

File: docs/extensions/idm/_index.md [hunk 4]
```diff
@@ -10,6 +10,20 @@ geekdocCollapseSection: true
...
+[boltdb](https://github.com/etcd-io/bbolt) file `idm/ocis.boltdb` inside the oCIS base data directory.
+
+Note: IDM is limited in its functionality. It only supports a subset of the LDAP operations (namely BIND, SEARCH, ADD, MODIFY, DELETE).
+Also IDM currently does not do any Schema Verification (e.g. structural vs. auxillary Objectclasses, require and option Attributes,
```
Comment (micbar) [inline docs/extensions/idm/_index.md:25]:
```suggestion
Also IDM currently does not do any Schema Verification (e.g. structural vs. auxiliary Objectclasses, require and option Attributes,
```

File: docs/extensions/idm/configuration_hints.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+geekdocEditPath: edit/master/docs/extensions/idm
+geekdocFilePath: configuration_hints.md
+geekdocCollapseSection: true
+---
+
+## TLS Server Certificates
+By default IDM generates a self-signed certificate and key on startup to be
+able to provide TLS protected services. The certificate is stored in
+`idm/ldap.crt` inside the oCIS base data directory. The key is in
+`idm/ldap.key` in the same directory. You configure custom a custom server
```
Comment (micbar) [inline docs/extensions/idm/configuration_hints.md:15]:
```suggestion
`idm/ldap.key` in the same directory. You can use a custom server
```

File: docs/extensions/idm/configuration_hints.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+certificate by setting the `IDM_LDAPS_CERT` and `IDM_LDAPS_KEY`.
+
+## Default / Demo Users
+On startup IDM creates a set of default services users, that are needed
```
Comment (micbar) [inline docs/extensions/idm/configuration_hints.md:19]:
```suggestion
On startup IDM creates a set of default services users that are needed
```

File: docs/extensions/idm/configuration_hints.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+internally to provide other oCIS service access to IDM. These users are stored
```
Comment (micbar) [inline docs/extensions/idm/configuration_hints.md:20]:
```suggestion
internally to provide access to IDM to other oCIS services. These users are stored
```

File: docs/extensions/idm/configuration_hints.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+in a separate subtree. The base DN of that subtree is:
+`ou=sysusers,o=libregraph-idm`. The service users are:
+
+* `uid=libregraph,ou=sysusers,o=libregraph-idm`: This is the only user with write
+  access to the LDAP tree. It is used by the Graph service to lookup, create, delete
```
Comment (micbar) [inline docs/extensions/idm/configuration_hints.md:25]:
```suggestion
  access to the LDAP tree. It is used by the Graph service to lookup, create, delete and
```

File: docs/extensions/idm/configuration_hints.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  modify users and groups.
+* `uid=idp,ou=sysusers,o=libregraph-idm`: This user is used by the IDP service to
+  perform user lookups for authentication.
+* `uid=reva,ou=sysusers,o=libregraph-idm`: This user is used by the "reva" services
+  "user, group and auth-basic.
```
Comment (micbar) [inline docs/extensions/idm/configuration_hints.md:30]:
```suggestion
  `user`, `group` and `auth-basic`.
```

File: docs/extensions/idm/configuration_hints.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+IDM is also able to create [Demo Users](../../../ocis/getting-started/demo-users)
+upon startup. 
+
+## Access via LDAP command line tools
+For testing purposes it is sometimes helpful to query IDM using the ldap
+command line clients. To e.g. list all user can use this command:
```
Comment (micbar) [inline docs/extensions/idm/configuration_hints.md:37]:
```suggestion
command line clients. To e.g. list all users, this command can be used:
```

File: docs/extensions/idm/configuration_hints.md [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+```
+ldapsearch -x -H ldaps://127.0.0.1:9235 -x -D uid=libregraph,ou=sysusers,o=libregraph-idm -w idm -b o=libregraph-idm objectclass=inetorgperson
+```
+
+When using the default configuration with the self-signed server certificate
+you might need to switch of Certificate Validation the `LDAPTL_REQCERT` env
```
Comment (micbar) [inline docs/extensions/idm/configuration_hints.md:44]:
```suggestion
you might need to switch off the Certificate Validation using `LDAPTL_REQCERT` env
```

**Review by micbar — APPROVED** [review]
Please feel free to commit my suggestions and merge.

**Comment by micbar** [comment]
@mmattel FYI

File: docs/extensions/idm/configuration_hints.md [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+date: 2022-04-27:00:00+00:00
+weight: 20
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/extensions/idm
+geekdocFilePath: configuration_hints.md
+geekdocCollapseSection: true
+---
+
+## TLS Server Certificates
+By default IDM generates a self-signed certificate and key on startup to be
```
Comment (mmattel) [inline docs/extensions/idm/configuration_hints.md:12]:
```suggestion
By default IDM generates a self-signed certificate and key on first startup to be
```

**Pull request merged**